ConwaysGameOfLife/Universe.cs

```diff
@@ -65,7 +71,7 @@ public List<Location> GetCellNeighbourLocations(Cell cell)
         public int CountLiveNeighbours(Cell cell)
         {
             var aliveNeighbours = new List<Location>();
-            ThrowsExceptionIfCellIsNull(cell);

```

Comment:
I'm not a fan of throwing random unhandled exceptions. There are more graceful ways of handling stuff like this e.g. simply not passing any potential nulls to `CountLiveNeighbours`